teacher.rb

```diff
@@ -0,0 +1,18 @@
+require_relative './person'
+
+class Teacher < Person
+  attr_accessor :specialization
+
+  def initialize(
+    specialization,
+    age,
+    name = 'Unknown'
+  )
+    super(age, name, true)
+    @specialization = specialization
+  end
```

Comment:
- [ ] Kindly add an paratmeter parent_permission: true in your contstructor and pass it to the parent class 🤔 I know it sounds confusing but you should follow the requirements 👍

**Your code will look something similar to the below ⬇️ code**

```ruby
def initialize(age, specialization, name = "Unknown", parent_permission = true)
    super(age, name, parent_permission)
    @specialization = specialization
  end
```

Reply:
Hi @HAMMAS-SALEEM 

Are you sure about your requirement?
The teacher does not need the parent permission parameter to be specified and I sent it to the parent class using super to be always true.